sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -253,6 +253,41 @@ case class GpuProjectAstExec(
   override def outputBatching: CoalesceGoal = GpuExec.outputBatching(child)
 }
 
+/**
+ * Do projections in a tiered fashion, where earlier tiers contain sub-expressions that are
+ * referenced in later tiers.  Each tier adds columns to the original batch corresponding
+ * to the output of the sub-expressions.
+ */
```

Comment:
it would be helpful, if there could be minimal example in this header that shows the projected tier process for a couple of tiers and a simple original `cb`. It took me a while to grok how this function works.

Reply:
I will add a comment.

Reply:
Comment added.  I also added a unit test like what the comment says.

Reply:
Thanks!